Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,280 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class EventService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+﻿using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Exceptions;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using System.Transactions;
+using System.Text.RegularExpressions;
+using ReasnAPI.Mappers;
+
+namespace ReasnAPI.Services;
+public class EventService(ReasnContext context, ParameterService parameterService, TagService tagService)
+{
+
+    public EventDto CreateEvent(EventDto eventDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var createdTime = DateTime.UtcNow;
+            var newEvent = eventDto.ToEntity();
+            newEvent.CreatedAt = createdTime;
+            newEvent.UpdatedAt = createdTime;
+            newEvent.Slug = CreateSlug(eventDto, createdTime);
+
+            context.Events.Add(newEvent);
+
+            context.SaveChanges();
+            if (eventDto.Tags != null)
+            {
+                var newTags = eventDto.Tags.ToEntityList();
+                AttatchTagsToEvent(newTags, newEvent);
+            }
+
+            if (eventDto.Parameters != null)
+            {
+                var newParameters = eventDto.Parameters.ToEntityList();
+                AttachParametersToEvent(newParameters, newEvent);
+            }
+
+            context.SaveChanges();
+            scope.Complete();
+        }
+
+        return eventDto;
+    }
+
+    public EventDto UpdateEvent(int eventId, EventDto eventDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var eventToUpdate = context.Events.Include(e => e.Tags).Include(e => e.Parameters).FirstOrDefault(e => e.Id == eventId);
+
+            if (eventToUpdate is null)
+            {
+                throw new NotFoundException("Event not found");
+            }
+
+            eventToUpdate.Name = eventDto.Name;
+            eventToUpdate.AddressId = eventDto.AddressId;
+            eventToUpdate.Description = eventDto.Description;
+            eventToUpdate.OrganizerId = eventDto.OrganizerId;
+            eventToUpdate.StartAt = eventDto.StartAt;
+            eventToUpdate.EndAt = eventDto.EndAt;
+            eventToUpdate.UpdatedAt = DateTime.Now;

```

Comment:
Datetime.UtcNow, not date without offset

Reply:
changed